### PR TITLE
[RFC] Allows creating files from folders list

### PR DESCRIPTION
**Pull request by shtanton**
For issue #1022 
Adds 'newfile' action on folders list

### CODE REVIEW

**Comment by shtanton** [comment]
I'm not sure if you want tests for the new action, I don't know how to add them as the only existing test for folders seems to be checking the delete action runs, but not checking it deletes anything.

If you want some unit tests I'll see if I can figure it out

File: src/list/source/folders.ts [hunk 1]
```diff
@@ -25,6 +30,18 @@ export default class FoldList extends BasicList {
     this.addAction('delete', async item => {
       workspace.removeWorkspaceFolder(item.label)
     }, { reload: true, persist: true })
+
+		this.addAction('newfile', async item => {
+			let file = await nvim.call('input', ['File name:', item.label + '/'])
```
Comment (chemzqm) [inline src/list/source/folders.ts]: You have to use `workspace.requestInput` api instead,  this call won't work on vim8.

Comment (shtanton) [reply src/list/source/folders.ts]: Cool, I've fixed that

File: src/list/source/folders.ts [hunk 2]
```diff
@@ -25,6 +30,18 @@ export default class FoldList extends BasicList {
...
+		this.addAction('newfile', async item => {
+			let file = await workspace.requestInput('File name:', item.label + '/')
+			let dir = path.dirname(file)
+			if (!fs.existsSync(dir)) {
+				mkdirp.sync(dir)
+			}
+			await workspace.createFile(file, {overwrite: false, ignoreIfExists: true})
+			let range = Range.create(0, 0, 0, 0)
+			let location = Location.create(URI.file(file).toString(), range)
+			await this.jumpTo(location)
```
Comment (chemzqm) [inline src/list/source/folders.ts]: No need for location, Just use `await this.jumpTo(URI.file(file).toString())`

Comment (shtanton) [reply src/list/source/folders.ts]: Thanks, updated that

**Pull request merged**